smithy-aws-protocol-tests/model/aws-config.smithy

```diff
@@ -0,0 +1,95 @@
+// This file defines standard AWS configuration options that may be needed for
+// testing certain behaviors. This isn't an exhaustive listing of all possible
+// configuration settings. See the following for a more comprehensive overview:
+// https://docs.aws.amazon.com/credref/latest/refdocs/overview.html
+
+$version: "1.0"
+
+metadata suppressions = [{
+    id: "UnreferencedShape",
+    namespace: "aws.protocoltests.config",
+    reason: """
+        These shapes are intended to be used to validate vendorParams in
+        protocol tests, so they naturally will not be connected to a service.
+        """
+}]
+
+namespace aws.protocoltests.config
+
+structure AwsConfig {
+    /// This is the time that should be set during the course of the test.
+    /// This is important for things like signing where the clock time impacts
+    /// the result.
+    clockTime: Timestamp,
+    scopedConfig: ScopedConfig,
+}
+
+/// Config settings that are scoped to different sources, such as environment
+/// variables or the AWS config file.
+structure ScopedConfig {
+    environment: EnvironmentConfig,
+    file: FileConfig,
+    client: ClientConfig,
+    operation: OperationConfig,
+}
+
+/// Config settings that can be set as environment variables.
+structure EnvironmentConfig {
+    AWS_ACCESS_KEY_ID: String,
+    AWS_SECRET_ACCESS_KEY: String,
+    AWS_DEFAULT_REGION: String,
+    AWS_RETRY_MODE: RetryMode,
+}
+
+/// Config settings that can be set in the AWS config file.
+structure FileConfig {
+    access_key_id: String,
```

Comment:
I'm not positive I would model these—by modeling this explicitly, we'd need to upgrade Smithy for any new features that wanted to be tested (which is maybe OK?)

Reply:
These are for the protocol tests here, you'd need to update this package anyway to add more tests to it. In any event, adding a line here is never gonna be the bottleneck.